lib/datadog/core/utils/safe_dup.rb

```diff
@@ -16,8 +16,23 @@ def dup
           using RefineNil
         end
 
-        def self.frozen_or_dup(v)
-          v.frozen? ? v : v.dup
+        # String#+@ was introduced in Ruby 2.3
+        if String.method_defined?(:+@) && String.method_defined?(:-@)
+          def self.frozen_or_dup(v)
+            v.frozen? ? v : +(-v)
```

Comment:
@nirebu, can you leave a comment here explaining why `+(-v)` is being done here? I think this will raise questions in the future when someone stumbles upon this line.

Reply:
@marcotc added!